### PR TITLE
Added starter workflow to help get started with EthicalCheck code-scanning Action

### PR DESCRIPTION
**Pull request by abdul-hai-apisec**
## Pre-requisites

- [x] Prior to submitting a new workflow, please apply to join the GitHub Technology Partner Program: [partner.github.com/apply](https://partner.github.com/apply?partnershipType=Technology+Partner).

---

### **Please note that at this time we are only accepting new starter workflows for Code Scanning. Updates to existing starter workflows are fine.**

---

## Tasks

**For _all_ workflows, the workflow:**

- [x] Should be contained in a `.yml` file with the language or platform as its filename, in lower, [_kebab-cased_](https://en.wikipedia.org/wiki/Kebab_case) format (for example, [`docker-image.yml`](https://github.com/actions/starter-workflows/blob/main/ci/docker-image.yml)).  Special characters should be removed or replaced with words as appropriate (for example, "dotnet" instead of ".NET").
- [x] Should use sentence case for the names of workflows and steps (for example, "Run tests").
- [x] Should be named _only_ by the name of the language or platform (for example, "Go", not "Go CI" or "Go Build").
- [x] Should include comments in the workflow for any parts that are not obvious or could use clarification.
- [x] Should specify least priviledge [permissions](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#modifying-the-permissions-for-the-github_token) for `GITHUB_TOKEN` so that the workflow runs successfully.

**For _Code Scanning_ workflows, the workflow:**

- [x] Should be preserved under [the `code-scanning` directory](https://github.com/actions/starter-workflows/tree/main/code-scanning).
- [x] Should include a matching `code-scanning/properties/*.properties.json` file (for example, [`code-scanning/properties/codeql.properties.json`](https://github.com/actions/starter-workflows/blob/main/code-scanning/properties/codeql.properties.json)), with properties set as follows:
  - [x] `name`: Name of the Code Scanning integration.
  - [x] `organization`: Name of the organization producing the Code Scanning integration.
  - [x] `description`: Short description of the Code Scanning integration.
  - [x] `categories`: Array of languages supported by the Code Scanning integration.
  - [x] `iconName`: Name of the SVG logo representing the Code Scanning integration. This SVG logo must be present in [the `icons` directory](https://github.com/actions/starter-workflows/tree/main/icons).
- [x] Should run on `push` to `branches: [ $default-branch, $protected-branches ]` and `pull_request` to `branches: [ $default-branch ]`. We also recommend a `schedule` trigger of `cron: $cron-weekly` (for example, [`codeql.yml`](https://github.com/actions/starter-workflows/blob/c59b62dee0eae1f9f368b7011cf05c2fc42cf084/code-scanning/codeql.yml#L14-L21)).

**Some general notes:**

- [x] This workflow must _only_ use actions that are produced by GitHub, [in the `actions` organization](https://github.com/actions), **or**
- [x] This workflow must _only_ use actions that are produced by the language or ecosystem that the workflow supports.  These actions must be [published to the GitHub Marketplace](https://github.com/marketplace?type=actions).  We require that these actions be referenced using the full 40 character hash of the action's commit instead of a tag.  Additionally, workflows must include the following comment at the top of the workflow file:
    ```
    # This workflow uses actions that are not certified by GitHub.
    # They are provided by a third-party and are governed by
    # separate terms of service, privacy policy, and support
    # documentation.
    ```
- [x] Automation and CI workflows should not send data to any 3rd party service except for the purposes of installing dependencies.
- [x] Automation and CI workflows cannot be dependent on a paid service or product.

### CODE REVIEW

File: code-scanning/ethicalcheck.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+       - name: EthicalCheck  Free & Automated API Security Testing Service
+         uses: apisec-inc/ethicalcheck-action@latest
+         with:
+          # The OpenAPI Specification URL or Swagger Path or Public Postman collection URL.
+          oas-url: "http://netbanking.apisec.ai:8080/v2/api-docs"
+          # The email address to which the penetration test report will be sent.
+          email: "xxx@apisec.ai"
```
Comment (leftrightleft) [inline code-scanning/ethicalcheck.yml:57]: Please add a new line at the end of a yml file

Comment (leftrightleft) [inline code-scanning/ethicalcheck.yml:57]: There are no steps to upload SARIF to code scanning in this workflow.  Take a look at [our documentation](https://docs.github.com/en/code-security/code-scanning/integrating-with-code-scanning/uploading-a-sarif-file-to-github) for uploading SARIF.  Here's an example from the other [APISec starter workflow](https://github.com/actions/starter-workflows/blob/main/code-scanning/apisec-scan.yml#L66-L69).

File: code-scanning/ethicalcheck.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  workflow_dispatch:
+
+
+jobs:
+  Trigger_EthicalCheck:
+    runs-on: ubuntu-latest
+
+    steps:
+       - name: EthicalCheck  Free & Automated API Security Testing Service
+         uses: apisec-inc/ethicalcheck-action@latest
```
Comment (leftrightleft) [inline code-scanning/ethicalcheck.yml:52]: When pointing to an action in a starter workflow, please reference the SHA of the Actions release you would like your customers to use.  This way, we can be sure that the action contains the code you expect.  For example, the latest release of this action would be called like this:
```
uses: apisec-in/ethicalcheck-action@3ec5e93b42e591349e46635da9f909bac66c23a9
```
Here's a screenshot that shows you were to get this SHA from:
image

**Review by leftrightleft — CHANGES_REQUESTED** [review]
A few changes are needed before this can be merged:
1. Starter workflows are for integrating with Code Scanning via SARIF.  This workflow doesn't have the ability to upload (or generate) SARIF documents.
2. We always need to point to a specific SHA when referencing an Action.  

Take a look at the comments inline they capture the next steps for you 👍

File: code-scanning/ethicalcheck.yml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  push:
+    branches: [ $default-branch, $protected-branches ]
+  pull_request:
+    branches: [ $default-branch ]
+  schedule:
+    - cron: $cron-weekly
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
```
Comment (Phantsure) [inline code-scanning/ethicalcheck.yml:44]: Kindly add workflow level as well as job level permissions

File: code-scanning/ethicalcheck.yml (hunk 1 above)
Comment (abdul-hai-apisec) [reply code-scanning/ethicalcheck.yml:57]: As of today we are only sending the email report with ethical check github action, With regards to integrating it with code scanning using SARIF, we will comeback in 4-7 weeks.

Comment (abdul-hai-apisec) [reply code-scanning/ethicalcheck.yml]: Done

Comment (abdul-hai-apisec) [reply code-scanning/ethicalcheck.yml]: Added a step to upload SARIF

File: code-scanning/ethicalcheck.yml (hunk 2 above)
Comment (abdul-hai-apisec) [reply code-scanning/ethicalcheck.yml]: Added reference to SHA

File: code-scanning/ethicalcheck.yml (hunk 3 above)
Comment (Phantsure) [reply code-scanning/ethicalcheck.yml:44]: @abdul-hai-apisec Can we have this?

**Comment by Phantsure** [comment]
CI check is failing. Can you look into that?

**Comment by abdul-hai-apisec** [comment]
> CI check is failing. Can you look into that?

Fixed

**Comment by Phantsure** [comment]
@abdul-hai-apisec I would like to take this ahead. Can you merge main in this branch?
@leftrightleft Hope this looks fine?

**Review by Phantsure — APPROVED** [review]

File: code-scanning/properties/ethicalcheck.properties.json [hunk 4]
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "EthicalCheck",
+  "creator": "APIsec",
+  "description": "EthicalCheck addresses the critical need to continuously security test APIs in development and in production. EthicalCheck provides the industry’s only free & automated API security testing service that uncovers security vulnerabilities using OWASP API list. Developers relies on EthicalCheck to evaluate every update and release, ensuring that no APIs go to production with exploitable vulnerabilities. You develop the application and API, we bring complete and continuous security testing to you, accelerating development.Know your API and Applications are secure with EthicalCheck – our free & automated API security testing service.",
```
Comment (Phantsure) [inline code-scanning/properties/ethicalcheck.properties.json:4]: This description is too big. Can we make it smaller to fit in 3 lines here:
image

Comment (abdul-hai-apisec) [reply code-scanning/properties/ethicalcheck.properties.json:4]: Created pull request for same. https://github.com/actions/starter-workflows/pull/1787

**Pull request merged**